analyze_test.go

```diff
@@ -1067,3 +1067,210 @@ func TestAnalyze_nap_app_protect_reconnect_period_seconds(t *testing.T) {
 		})
 	}
 }
+
+//nolint:funlen
+func TestAnalyze_http3(t *testing.T) {
+	t.Parallel()
+	testcases := map[string]struct {
+		stmt    *Directive
+		ctx     blockCtx
+		wantErr bool
+	}{
+		"http3 ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"on"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3 not ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"somevalue"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			true,
+		},
+		"http3_hq ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"on"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3_hq not ok": {
+			&Directive{
+				Directive: "http3",
+				Args:      []string{"somevalue"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			true,
+		},
+		"http3_max_concurrent_streams ok": {
+			&Directive{
+				Directive: "http3_max_concurrent_streams",
+				Args:      []string{"10"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3_max_concurrent_streams not ok": {
+			&Directive{
+				Directive: "http3_max_concurrent_streams",
+				Args:      []string{"10"},
+				Line:      5,
+			},
+			blockCtx{"http", "location"},
+			true,
+		},
+		"http3_stream_buffer_size ok": {
+			&Directive{
+				Directive: "http3_stream_buffer_size",
+				Args:      []string{"128k"},
+				Line:      5,
+			},
+			blockCtx{"http", "server"},
+			false,
+		},
+		"http3_stream_buffer_size not ok": {
+			&Directive{
+				Directive: "http3_stream_buffer_size",
+				Args:      []string{"128k"},
+				Line:      5,
+			},
+			blockCtx{"http", "location"},
+			true,
+		},
+	}
+
+	for name, tc := range testcases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			err := analyze("nginx.conf", tc.stmt, ";", tc.ctx, &ParseOptions{})
+
+			if !tc.wantErr && err != nil {
```

Comment:
This logic is a bit convoluted in my mind:

Suggestion:
```
if tc.wantError{
    require.Error(err)
} else {
    require.Nil(erro)
}
```

Reply:
Not sure if I should change this check, as other tests in this file are using this pattern